solidity/contracts/midas/MidasAllocator.sol

```diff
@@ -104,7 +116,18 @@ contract MidasAllocator is IDispatcher, Maintainable {
     ///         Acre Vault contract.
     /// @dev This is a special function that can be used to migrate funds during
     ///      allocator upgrade or in case of emergencies.
-    function emergencyWithdraw() external onlyOwner {
-        revert("not implemented");
+    function emergencyWithdraw() external onlyOwner returns (uint256) {
+        uint256 shares = vaultSharesToken.balanceOf(address(this));
+        vaultSharesToken.forceApprove(address(midasVault), shares);
+        return midasVault.requestRedeem(shares, address(acreVault));
+    }
+
+    /// @notice Sets the withdrawal queue address.
+    /// @param _withdrawalQueue Address of the withdrawal queue.
+    function setWithdrawalQueue(address _withdrawalQueue) external onlyOwner {
+        if (_withdrawalQueue == address(0)) {
+            revert ZeroAddress();
+        }
```

Comment:
In acreBTC, the withdrawal queue can be zero address to disable withdrawals. Why not allowing it to be zero here as well, for consistency? The transfer in `withdrawShares` is not going to work as a zero address can't call it.